pkgs/development/python-modules/glcontext/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, isPy3k
+, libGL
+, libX11
+, pytestCheckHook
+, psutil
+}:
+
+buildPythonPackage rec {
+  pname = "glcontext";
+  version = "2.3.3";
+
+  src = fetchFromGitHub {
+    owner = "moderngl";
+    repo = pname;
+    rev = version;
+    sha256 = "16kwrfjijn9bnb48rk17wapmhxq6g9s59zczh65imyncb9k82wkc";
+  };
+
+  disabled = !isPy3k;
+
+  buildInputs = [ libGL libX11 ];
+
+  # tests fail because they try to open display => doCheck = false.
```

Comment:
Does it work if you set LD_LIBRARY_PATH in `preCheck` instead of disabling the check phase?

Reply:
No, i already tried this. Also it seems like setting LD_LIBRARY_PATH in preCheck doesn't make it to actual checkPhase. When i do that and uncomment checkPhase pytest fails with
```
E       ImportError: cannot import name 'x11' from 'glcontext' (/build/source/glcontext/__init__.py)
```
When i set checkPhase like i did in my commit pytest fails with
```
E       Exception: (standalone) XOpenDisplay: cannot open display
```
The only way to give the included tests a chance would be to provide access display during build. AFAIK it is not possible so far. Still, I created a nix-shell with this `shell.nix` :
``` nix
{ pkgs ? import ./. {} }: with pkgs;
let
  pythonEnv = python38.withPackages (ps: [
      ps.glcontext
      ps.python
      ps.psutil
      ps.pytest
  ]);
in mkShell {
  nativeBuildInputs = [
    pythonEnv
    libGL
    xorg.libX11
  ];
  shellHook = ''
    export LD_LIBRARY_PATH=${libGL}/lib:${xorg.libX11}/lib:$LD_LIBRARY_PATH
  '';
}
```
at the root of my nixpkgs-repo & with glcontext's source dir unpacked through unpackPhase. Like this running `pytest tests/` inside source-dir all tests succeeded. Is there a way to turn this in to a test that works in checkPhase?

Another possibility would be to write some nix-specific tests which don't rely on displays. But i don't know if it makes any sense  for this package.

Reply:
I think it's fine to disable tests given that they require a display. If this were a highly-used package, you could write a NixOS test for it (running in a VM and all) but it doesn't make sense to do that in this case.

Either way, I think settig `doCheck` to `false` implies that `checkPhase` does not run even if you explicitly set it, so `pytest` is not being run. If you confirm that's the case, I think adding a comment pointing to this discussion is better than having a dummy`checkPhase`.